File: next-greater-element-i/next-greater-element-i.cpp
class Solution {

int search(vector<int>nums,int k, int ele){
    for(int i=k;i<nums.size();i++){
        if(nums[i]>ele)
        return nums[i];
    }
    return -1;
}


public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int>ans;
        
        for(int i=0;i<nums1.size();i++){
            int ele = nums1[i];
            for(int j=0;j<nums2.size();j++){
                if(ele==nums2[j]){
                int k = search(nums2,j,ele);
                ans.push_back(k);
                break;
                }
            }
        }
        return ans;

    }
};

File: 496-next-greater-element-i/next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        
        
        int[] ans = new int[nums1.length];

        // stack and map 

        Stack<Integer> st = new Stack<>();
        HashMap<Integer,Integer> mp = new HashMap<>();


        for(int num:nums2){

            while(!st.empty() && num>st.peek()){
                mp.put(st.pop(),num);
            }

            st.add(num);

        }

        int i = 0;

        for(int num:nums1){
         ans[i++] = mp.getOrDefault(num,-1);
        }


        return ans;
    }
}